### PR TITLE
Fixes and improvements

### PR DESCRIPTION
**Pull request by AntonD3**
# What :computer: 
* Increased rich wallets number to 10
* Increased rich wallets start balance
* Fixed `eth_getCode`
* Fixed revert processing in the `eth_call`
* Fixed reverts in the `eth_estimateGas`
* Fixed `effective_gas_price` in the `eth_getTransactionReceipt`

# Why :hand:
* Some complex tests require more wallets, for example, DEFI, where needed to be separate accounts for the liquidity providers, traders, owners, etc.
* A big amount of ether can be used when you want to write tests with real-world amounts
* Previously `eth_code` was returning `codehash`, not the code, it was fixed
* The error in case of transaction revert during `eth_call` should be returned to be compatible with the testing tools
* The error data returned from `eth_estimateGas` was updated, to be compatible as well
* The wrong constant value for the `effective_gas_price` was used

# Evidence :camera:
Screenshot 2023-08-18 at 10 25 49

[Here is](https://github.com/matter-labs/hardhat-zksync/blob/main/packages/hardhat-zksync-chai-matchers/src/internal/reverted/utils.ts#L14) an example where error data is used by a tooling

[Here is](https://github.com/TrueFiEng/Waffle/blob/master/waffle-chai/src/matchers/reverted.ts#L8) - `revert` word in the error body/message

BEFORE and AFTER of `eth_code` request:
image

Before:
image

After:
image

# Notes 📝 
* This addresses https://github.com/matter-labs/era-test-node/issues/33

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -288,7 +312,7 @@ async fn main() -> anyhow::Result<()> {
         let address = wallet.0;
         let private_key = wallet.1;
         node.set_rich_account(H160::from_str(address).unwrap());
-        println!("Account #{}: {} (10000 ETH)", index, address);
+        println!("Account #{}: {} (1000000000000 ETH)", index, address);
```
Comment (MexicanAce) [inline src/main.rs:315]: As this is now a much larger number, please use underscore notation for ease of readability: e.g. `1_000_000_000_000`

File: src/node.rs [hunk 2]
```diff
@@ -1083,7 +1121,7 @@ impl EthNamespaceT for InMemoryNode {
                 } else {
                     U64::from(0)
                 }),
-                effective_gas_price: Some(500.into()),
+                effective_gas_price: Some(250_000_000.into()),
```
Comment (MexicanAce) [inline src/node.rs:1124]: Should we move this to a `public const` to the top of the file?

**Review by MexicanAce — CHANGES_REQUESTED** [review]
Please add Verification Evidence to the PR description for all changes. Also, I think this resolved 2 of the open GitHub Issues, so let's tag them in the Notes section of the PR description as well.

File: src/main.rs (hunk 1 above)
Comment (AntonD3) [reply src/main.rs]: Done

File: src/main.rs [hunk 3]
```diff
@@ -101,6 +101,30 @@ pub const RICH_WALLETS: [(&str, &str); 4] = [
         "0xA13c10C0D5bd6f79041B9835c63f91de35A15883",
         "0x850683b40d4a740aa6e745f889a6fdc8327be76e122f5aba645a5b02d0248db8",
     ),
+    (
+        "0x8002cD98Cfb563492A6fB3E7C8243b7B9Ad4cc92",
```
Comment (MexicanAce) [inline src/main.rs:105]: Let's add these 7 new accounts to the `README.md` table

**Review by MexicanAce — APPROVED** [review]
LGTM

**Pull request merged**